packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts

```diff
@@ -155,117 +156,239 @@ export class AzureCommunicationCallAdapter implements CallAdapter {
     this.callClient.onStateChange(onStateChange);
   }
 
-  public dispose(): void {
+  public dispose = async (): Promise<void> => {
     this.callClient.offStateChange(this.onClientStateChange);
-    this.callAgent.dispose();
-  }
 
-  public isTeamsCall(): boolean {
+    return withErrorHandling(
+      this.context,
+      this.emitter,
+      async () => {
+        await this.callAgent.dispose();
+      },
+      Promise.resolve()
+    );
+  };
+
+  public isTeamsCall = (): boolean => {
     return 'meetingLink' in this.locator;
-  }
+  };
 
-  public queryCameras(): Promise<VideoDeviceInfo[]> {
-    return this.deviceManager.getCameras();
-  }
+  public queryCameras = (): Promise<VideoDeviceInfo[]> => {
+    return withErrorHandling(
```

Comment:
Consider adding private methods to `AzureCommunicationCallAdapter` and `AzureCommunicationChatAdapter` to further redue boilerplate:

```
return this.handleAsyncError(
    async () => {
        return await this.deviceManager.getMicrophones();
    },
    []
)
```

Possibly adding `handleSyncError` as well if necessary.

Changes recommended:
* Naming: `withErrorHandling` sounds like a decorator to me. This is just a function call.
* Add a private method that can pull out `this.context` and `this.emitter` to reduce boilerplate.
* Make the generic method(s) smarter so it can infer the return type.

Reply:
Thanks for the feedback, I've refactored this as per your requirements, please take a look if it works for you.